discover/portal/articles/130-forms/06-data-providers.markdown

```diff
@@ -160,14 +160,20 @@ since a second call to the REST service provider is unnecessary.
 request, if a response is not returned.
 
 **Inputs**
-: Configure parameters (path parameter or query parameter) from the REST service to use to filter the REST service's response (displayed in the Outputs parameter configured below). Specify the
-Label, Parameter, and Type (Text or Number), and Choose whether the input is
-required to use the Data Provider. You can add multiple Inputs. These are used
-in Form Rules. A User enters input into one field, and their input is sent to
-the REST service. The REST service's response data is filtered by the input.
+: Configure path or query parameters from the REST service to filter the REST
+service's response. Specify the Label, Parameter, and Type (Text or Number), and
+Choose whether the input is required to use the Data Provider. You can add
+multiple Inputs. To provide a way for users to specify the input value, use an
+[_Autofill_ Form Rule](/discover/portal/-/knowledge_base/7-1/action-autofill).
+A User enters input into one field, and their input is sent to the REST service.
+The REST service's response data is filtered by the input parameter.
 
 **Outputs**
-: The Parameter to display in Select from List or Text fields, the Text field only if the Autocomplete setting is enabled. You can add multiple Outputs. Outputs can be filtered by inputs (see above) but can also be displayed without configuring input filtering. Specify the Label, Path, and Type (Text,
-Number, or List). The Path field is specified in JsonPath syntax.
-
-![Figure 3: Set up Data Providers to display data retrieved from a REST service.](../../images/forms-data-provider-configuration.png)
\ No newline at end of file
+: The Parameter to display in Select from List or Text fields with autocomplete
+enabled. You can add multiple Outputs. Outputs can be filtered by inputs (see
+above) but can also be displayed without configuring input filtering. Specify
+the Label, Path, and Type (Text, Number, or List). The Path field is specified
+in JsonPath syntax. Using the `restcountries.eu` service, specify the `name`
```

Comment:
Yeah let's link to a good source. Do you have one? Rui gave us three links in his email from long ago, but it'd be nice to know which JsonPath implementation we're using in Liferay, so we can link to the appropriate docs.

Rui's links:
http://jsonpath.com/

http://goessner.net/articles/JsonPath/

https://www.pluralsight.com/blog/tutorials/introduction-to-jsonpath

When I type "JsonPath" into google, this is the first results:

https://github.com/json-path/JsonPath

Reply:
Okay, I found this import statement in DDMRESTDataProvider.java:

```import com.jayway.jsonpath.JsonPath;```

I'll try to find a better source of documentation than their github README, but it will work if we need it.

Reply:
I've asked for the guys of Forms in order to share it with you. 😉